tox.ini

```diff
@@ -14,15 +14,15 @@ setenv =
 deps =
     -r{toxinidir}/test-requirements.txt
     -r{toxinidir}/requirements.txt
-    -c{env:UPPER_CONSTRAINTS_FILE:https://releases.openstack.org/constraints/upper/master}
 
 install_commands =
     pip install {opts} {packages}
 
 
 [testenv:{py3,py38,py39,py310}]
 basepython = python3
-deps = -r{toxinidir}/test-requirements.txt
+deps = -r{toxinidir}/requirements.txt
+       -r{toxinidir}/test-requirements.txt
```

Comment:
I think this is not necessary for this change thier.

Reply:
we need tenacity and openstack modules in py3 test